src/message/connection-endpoint.js

```diff
@@ -488,4 +488,20 @@ ConnectionEndpoint.prototype._isHttpsServer = function( ) {
 	return isHttps;
 };
 
+/**
+* Checks for authentication data and throws if null or not well formed
+*
+* @throws Will throw an error on invalid auth data
+*
+* @private
+* @returns {void}
+*/
+ConnectionEndpoint.prototype._getValidAuthData = function( authData ) {
+	var parsedData = JSON.parse( authData );
+	if( parsedData === null || parsedData === undefined ) {
```

Comment:
Should we add a type object check as  well? Otherwise you can login with a number or string which would be a bit wierd